generatebundlefile/ecr_test.go

```diff
@@ -0,0 +1,240 @@
+			dest:          fmt.Sprintf("%s/go/src/github.com/modelrocket/eks-anywhere-packages/generatebundlefile/hello-eks-anywhere", os.Getenv("HOME")),
+			wantErr:       false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			err := UnTarHelmChart(tc.testChartPath, tc.testChartName, tc.dest)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("UnTarHelmChart() error = %v, wantErr %v", err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestShaExistsInRepository(t *testing.T) {
+	tests := []struct {
+		testName       string
+		testRepository string
+		testVersion    string
+		checkPass      bool
+		wantErr        bool
+	}{
+		{
+			testName:       "Test empty Repository",
+			testRepository: "",
+			testVersion:    "sha256:0526725a65691944e831add6b247b25a93b8eeb1033dddadeaa089e95b021172",
+			wantErr:        true,
+		},
+		{
+			testName:       "Test empty Version",
+			testRepository: "hello-eks-anywhere",
+			testVersion:    "",
+			wantErr:        true,
+		},
+		{
+			testName:       "Test valid Repository and Version",
+			testRepository: "hello-eks-anywhere",
+			testVersion:    "sha256:0526725a65691944e831add6b247b25a93b8eeb1033dddadeaa089e95b021172",
+			checkPass:      true,
+			wantErr:        false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			ecrPublicClient, err := NewECRPublicClient(true)
+			if err != nil {
+				t.Fatalf("ecrPublicClient() did not work, %v", err)
+			}
+			got, err := ecrPublicClient.shaExistsInRepository(tc.testRepository, tc.testVersion)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("shaExistsInRepository() error = %v, wantErr %v", err, tc.wantErr)
+			}
+			if got != tc.checkPass {
+				t.Fatalf("shaExistsInRepository() = %#v\n\n\n, want %#v", got, tc.checkPass)
+			}
+		})
+	}
+}
+
+func TestTagFromSha(t *testing.T) {
+	tests := []struct {
+		testName       string
+		testRepository string
+		testDigest     string
+		checkVersion   string
+		wantErr        bool
+	}{
+		{
+			testName:       "Test empty Repository",
+			testRepository: "",
+			testDigest:     "sha256:0526725a65691944e831add6b247b25a93b8eeb1033dddadeaa089e95b021172",
+			wantErr:        true,
+		},
+		{
+			testName:       "Test empty Version",
+			testRepository: "hello-eks-anywhere",
+			testDigest:     "",
+			wantErr:        true,
+		},
+		{
+			testName:       "Test valid Repository and Version",
+			testRepository: "hello-eks-anywhere",
+			testDigest:     "sha256:0526725a65691944e831add6b247b25a93b8eeb1033dddadeaa089e95b021172",
+			checkVersion:   "v0.1.1-baa4ef89fe91d65d3501336d95b680f8ae2ea660",
+			wantErr:        false,
+		},
+	}
+	//images.Digest, err = ecrClient.tagFromSha(images.Repository, images.Digest)
+	for _, tc := range tests {
+		t.Run(tc.testName, func(tt *testing.T) {
+			ecrClient, err := NewECRClient(true)
+			if err != nil {
+				t.Fatalf("ecrClient() did not work, %v", err)
+			}
+			got, err := ecrClient.tagFromSha(tc.testRepository, tc.testDigest)
```

Comment:
Same here. Ideally we'd stub these out. There's a place for this kind of testing, but it's not in the main unit test battery.

Reply:
Added an issue for this.